internal/clients/http/image_builder/image_client.go

```diff
@@ -45,21 +44,20 @@ func (c *ibClient) Ready(ctx context.Context) error {
 	defer span.End()
 
 	logger := logger(ctx)
-	resp, err := c.client.GetReadiness(ctx, headers.AddImageBuilderIdentityHeader, headers.AddEdgeRequestIdHeader)
+	resp, err := c.client.GetReadinessWithResponse(ctx, headers.AddImageBuilderIdentityHeader, headers.AddEdgeRequestIdHeader)
 	if err != nil {
 		logger.Error().Err(err).Msg("Readiness request failed for image builder")
 		return err
 	}
-	defer func() {
-		if tempErr := resp.Body.Close(); tempErr != nil {
-			logger.Error().Err(tempErr).Msg("Readiness request for image builder: response body close error")
-		}
-	}()
```

Comment:
Why did you remove the defer?

Reply:
I have replaced the calls with `XXX**WithResponse**` function, as [Lukas recommended](https://github.com/RHEnVision/provisioning-backend/pull/655#discussion_r1315926042). The generated code takes care of closing the body and only returns fields we need that we do not have to close. :)